test/modules/message-spec.js

```diff
@@ -250,10 +270,44 @@ describe('Message class test', () => {
       assert.deepEqual(prefix, notEmptyPrefix);
     });
 
+    it('should return data according to settings [settings] [log caller prefix] [disabled]', () => {
+      const message = new Message();
+      const prefix = message.getPrefix({ LOG_CALLER_PREFIX: false });
+      assert(!{}.hasOwnProperty.call(prefix, 'module'));
+      assert(!{}.hasOwnProperty.call(prefix, 'function'));
+      assert(!{}.hasOwnProperty.call(prefix, 'project'));
+    });
+
+    it('should return data according to settings [settings] [log caller prefix] [enabled]', () => {
+      const message = new Message();
+      const prefix = message.getPrefix({ LOG_CALLER_PREFIX: true });
+      assert({}.hasOwnProperty.call(prefix, 'module'));
+      assert({}.hasOwnProperty.call(prefix, 'function'));
+      assert({}.hasOwnProperty.call(prefix, 'project'));
+    });
+
+    it('should use env prior to settings [env and settings] [log caller prefix] [enabled]', () => {
+      const message = new Message();
+      process.env.LOG_CALLER_PREFIX = 'true';
+      const prefix = message.getPrefix({ LOG_CALLER_PREFIX: false });
+      assert({}.hasOwnProperty.call(prefix, 'module'));
+      assert({}.hasOwnProperty.call(prefix, 'function'));
+      assert({}.hasOwnProperty.call(prefix, 'project'));
+    });
+
+    it('should use env prior to settings [env and settings] [log caller prefix] [disabled]', () => {
+      const message = new Message();
+      process.env.LOG_CALLER_PREFIX = 'false';
+      const prefix = message.getPrefix({ LOG_CALLER_PREFIX: true });
+      assert(!{}.hasOwnProperty.call(prefix, 'module'));
```

Comment:
`module`, `function` and `project` props are included in prefix object only if `LOG_CALLER_PREFIX` is `true`. So, in case it is `false` or some unhandled error occurred, there won't be any of those keys in the prefix. That is why, I just check if it is present.

Also I am not using `assert.equal` because it's too environment specific and the result may differ depending on the system.